ethexe/contracts/src/Middleware.sol

```diff
@@ -0,0 +1,237 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.26;
+
+import {EnumerableMap} from "@openzeppelin/contracts/utils/structs/EnumerableMap.sol";
+import {Time} from "@openzeppelin/contracts/utils/types/Time.sol";
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+
+import {Subnetwork} from "symbiotic-core/src/contracts/libraries/Subnetwork.sol";
+import {IVault} from "symbiotic-core/src/interfaces/vault/IVault.sol";
+import {IRegistry} from "symbiotic-core/src/interfaces/common/IRegistry.sol";
+import {IEntity} from "symbiotic-core/src/interfaces/common/IEntity.sol";
+import {IBaseDelegator} from "symbiotic-core/src/interfaces/delegator/IBaseDelegator.sol";
+import {INetworkRegistry} from "symbiotic-core/src/interfaces/INetworkRegistry.sol";
+import {IOptInService} from "symbiotic-core/src/interfaces/service/IOptInService.sol";
+
+import {MapWithTimeData} from "./libraries/MapWithTimeData.sol";
+
+// TODO: support slashing
+// TODO: implement election logic
+// TODO: implement forced operators removal
+// TODO: implement forced vaults removal
+// TODO: implement rewards distribution
+contract Middleware {
+    using EnumerableMap for EnumerableMap.AddressToUintMap;
+    using MapWithTimeData for EnumerableMap.AddressToUintMap;
+    using Subnetwork for address;
+
+    error ZeroVaultAddress();
+    error NotKnownVault();
+    error VaultWrongEpochDuration();
+    error UnknownCollateral();
+    error OperatorGracePeriodNotPassed();
+    error VaultGracePeriodNotPassed();
+    error NotVaultOwner();
+    error IncorrectTimestamp();
+    error OperatorDoesNotExist();
+    error OperatorDoesNotOptIn();
+
+    uint48 public immutable ERA_DURATION;
```

Comment:
Its lower camel case everywhere if Im not mistaken
```suggestion
    uint48 public immutable eraDuration;
```

Reply:
Also wdyt of implementing some "struct Config" and keep only one immutable value?

Reply:
struct Config will be at next PR

Reply:
will make renaming next PR